lib/RabbitMQTransport.js

```diff
@@ -31,40 +30,35 @@ class RabbitMQTransport {
    */
   write({ message, queue, exchange = '' }) {
     return this.connect()
-      .then(() => new Promise((resolve, reject) => {
-        const publishResult =
-          this.channel.publish(exchange, queue, Buffer.from(this.stringify(message)));
-        if (publishResult) {
-          return resolve();
-        }
-        return reject(Error('Error sending message on MQ'));
-      }))
+      .then(() => this.channel.publish(
+        exchange,
+        queue,
+        Buffer.from(this.stringify(message))))
       .catch((err) => {
         throw err;
       });
   }
 
   /**
-   * Conenct to RabbitMQ
+   * Connect to RabbitMQ
    * @returns {Promise} resolves on success
    * @private
    */
   connect() {
-    if (this.connection && this.channel) {
-      return Promise.resolve();
-    }
-    return this.amqpLib.connect(this.mqURI)
-      .then((connection) => {
-        this.connection = connection;
-        return this.connection.createConfirmChannel();
-      })
-      .then((pchannel) => {
-        this.channel = pchannel;
-        return Promise.resolve();
-      })
-      .catch((err) => {
-        throw err;
-      });
+    return new Promise((resolve, reject) => {
+      if (this.connection && this.channel) {
+        return resolve();
+      }
+
+      try {
+        this.connection = this.amqpLib.connect([this.mqURI]);
+        this.channel = this.connection.createChannel();
```

Comment:
Why have you moved away from confirmation channel to a regular one? Couldn't this cause a data loss?

Reply:
https://www.rabbitmq.com/confirms.html#publisher-confirms

Reply:
This library provides different API
I am going to add `waitForConnect` before publish
This library provides this method

Reply:
I investigated it, so this library saves all messages to buffer and even if it is not connected we are not going to lose our data, and then when a connection is stable it uses RabbitMQ API to confirm channel inside of instance in this.channel
We don't have `waitForConnect` because it uses only for unit testing inside of this library
In short, this case is covered by this library